dipy/viz/ui.py

```diff
@@ -79,6 +87,15 @@ def __init__(self):
         self.on_right_mouse_button_dragged = lambda i_ren, obj, element: None
         self.on_key_press = lambda i_ren, obj, element: None
 
+    def _setup(self):
+        """ Setup this UI component.
+
+        This is where you should create all your needed actors and sub UI
+        components.
+        """
+        msg = "Subclasses of UI must implement `_setup(self)`."
+        raise NotImplementedError(msg)
+
     def get_actors(self):
```

Comment:
I'm not sure I understand the relationship between `get_actors()` and `add_to_renderer()`. It seems that simple classes like `Rectangle2D` which are built with a single VTK actor return that actor from `get_actors()` and use the `add_to_renderer()` from `UI`. More complex classes which are composites of simple classes, like `LineSlider2D`, return nothing from `get_actors()` and instead override `add_to_renderer()`. But then there's `TextBox2D`, which is a complex composite class in that it is built on top of `TextBlock2D` instead of a VTK actor, but it uses `get_actors()` more like a simple class. Can we simplify this somehow, perhaps by forcing subclasses to implement `add_to_renderer()` instead of `get_actors()`? Or by forcing composite classes to return their constituent actors in `get_actors()` like `TextBox2D` does?